src/mantine-core/src/components/Text/Text.tsx

```diff
@@ -3,20 +3,20 @@ import {
   DefaultProps,
   MantineSize,
   MantineGradient,
-  MantineColor,
   PolymorphicComponentProps,
   PolymorphicRef,
   useMantineDefaultProps,
 } from '@mantine/styles';
 import { Box } from '../Box';
+import { MantineTextColor } from './Text.types';
 import useStyles from './Text.styles';
 
 export interface SharedTextProps extends DefaultProps {
   /** Predefined font-size from theme.fontSizes */
   size?: MantineSize;
 
-  /** Text color from theme */
-  color?: MantineColor;
+  /** Text color from theme or dimmed variant */
+  color?: MantineTextColor;
```

Comment:
There is no need to extract this type, it should not be exported or used anywhere else, it should be just `color?: 'dimmed' | MantineColor`

Reply:
It's good to have a separate type as you can use it to assign the text color prior to using it in `Text` (or `Text` based components like `Menu.Label` for example). If we use a simple union, we have code duplication which also makes it harder to maintain and spot errors. But I guess it can always be added later if you decide on it.
I'll refactor it to a union for now, then.